lib/hsa/CMakeLists.txt

```diff
@@ -1,9 +1,13 @@
+include(FindPkgConfig)
+
 ####################
 # HCC runtime (HSA implementation)
 ####################
 if (HAS_ROCM EQUAL 1)
 add_mcwamp_library_hsa(mcwamp_hsa mcwamp_hsa.cpp unpinned_copy_engine.cpp)
-target_link_libraries(mcwamp_hsa  unwind)
+pkg_check_modules(LIBUNWIND REQUIRED libunwind)
+# TODO: we should probably add required CFLAGS and LDFLAGS from pkgconfig
+target_link_libraries(mcwamp_hsa ${LIBUNWIND_LIBRARIES})
```

Comment:
> The imported library is created in the hcc-config.cmake file as well. This allows for the dependencies to be relocatable. Otherwise hardcoded paths are used in the mcwamp_hsa target. Downstream libraries would use these hardcoded paths, so its better to create an imported target, so that hardcoded paths are not used.

that goes back to my first question. configure path is only used during hcc building, i.e. no downstream projects should use, so the imported stuff can be dropped, right? I understand its purpose in _hcc-config.cmake.in_

> There is no PUBLIC keyword that can be used when using add_library. This is an interface-only library just to store the flags from pkgconfig needed to use libunwind.

ah right, I confused this with link_target_libraries